source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
What about just calling `signals.post_constraints_creation(*this, constraints)`
after line 1397 in core.cc (i.e. within setup_dofs, before `constraints.close()` is called?